planner/core/rule_partition_processor.go

```diff
@@ -444,7 +464,33 @@ func (l *listPartitionPruner) findUsedListPartitions(conds []expression.Expressi
 			}
 			used[partitionIdx] = struct{}{}
 		} else {
-			return l.fullRange, nil
+			if len(r.HighVal) != len(exprCols) || r.IsFullRange() {
+				return l.fullRange, nil
+			}
+			if r.LowVal[0].Kind() == types.KindMinNotNull {
+				r.LowVal[0] = types.GetMinValue(exprCols[0].GetType())
+			}
+			if r.HighVal[0].Kind() == types.KindMaxValue {
+				r.HighVal[0] = types.GetMaxValue(exprCols[0].GetType())
+			}
+			lvalue, isLNull, err := pruneExpr.EvalInt(l.ctx, chunk.MutRowFromDatums(r.LowVal).ToRow())
+			if err != nil {
+				return nil, err
+			}
+			rvalue, isRNull, err := pruneExpr.EvalInt(l.ctx, chunk.MutRowFromDatums(r.HighVal).ToRow())
+			if err != nil {
+				return nil, err
+			}
+			if r.LowExclude {
+				lvalue++
+			}
+			if !r.HighExclude {
+				rvalue++
+			}
```

Comment:
How about move those logic into `l.listPrune.LocateRange`?

Reply:
It seems that there is no `ctx` in `l.listPrune`.